packages/react-avatar/src/components/Avatar/Avatar.types.ts

```diff
@@ -1,21 +1,15 @@
-import * as React from 'react';
 import { PresenceBadge } from '@fluentui/react-badge';
-import type {
-  ComponentProps,
-  ComponentSlotProps,
-  ComponentState,
-  IntrinsicSlotProps,
-  SlotRenderFunction,
-} from '@fluentui/react-utilities';
+import type { ComponentProps, ComponentState, SlotNoChildren, Slot } from '@fluentui/react-utilities';
 
 export type AvatarSlots = {
-  root: Omit<IntrinsicSlotProps<'span'>, 'color'> & { children?: never };
+  root: SlotNoChildren<'span'>;
 
   /**
-   * This overidden in the component's props, it's only here to make `getSlots` work
-   * `img`  is an exception since it should never accept children, but can accept a children render function
+   * The Avatar's image.
+   *
+   * Usage e.g.: `image={{ src: '...' }}`
    */
-  image?: IntrinsicSlotProps<'img'>;
+  image?: SlotNoChildren<'img'>;
```

Comment:
I understand why it's done in this way, but this moves the responsibility to developers. To clarify, there is nothing that stops me from doing this `Slot<'img'>`.

It would be nice if we could exclude `img` & `input` from list of elements on `Slot`, for example.

Reply:
DX wise I think that's a much better approach. We can enforce correct behavior by restricting our type.

Reply:
That is an excellent point. I'll update it to just Do The Right Thing, and prevent children on any empty element type (https://developer.mozilla.org/en-US/docs/Glossary/Empty_element). And then remove `SlotNoChildren` entirely.